client/src/components/TVDetailsPage.js

```diff
@@ -37,14 +38,32 @@ const TVDetailsPage = props => {
     }
     setShow(updatedShow)
   }
-  
+
+  const addNewVote = (newVote, reviewId) => {
+    const currentReviewIndex = show.reviews.findIndex((review) => review.id === reviewId)
+    let reviewsCopy = [...show.reviews]
+    reviewsCopy[currentReviewIndex].votes = [...reviewsCopy[currentReviewIndex].votes, newVote]
+    setShow({
+      ...show,
+      reviews: reviewsCopy
+    })
+  }
+
+  let loggedInError
+  if(!props.userId) {
+    loggedInError = <p>You must log in to vote on reviews!</p>
+  }
```

Comment:
So all unauthenticated users will see this error? Im okay with this, but a slightly more optimal approach would be to only show this error if they attempt to vote on something.

Otherwise, simply dont show the vote buttons at all unless they are logged in (you can still show total upvotes and downvotes regardless of authentication)